src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -133,49 +102,37 @@
           [message/message-with-reactions message-data context])]))])
 
 (defn messages-list
-  [{:keys [chat
-           pan-responder
-           show-input?]}]
-  (let [{:keys [group-chat chat-id public? community-id admins]} chat
-        messages                                                 (rf/sub [:chats/raw-chat-messages-stream
-                                                                          chat-id])
-        bottom-space                                             15]
+  [{:keys [chat-id] :as chat}]
+  (let [render-data  (rf/sub [:chats/current-chat-message-list-view-context])
+        messages     (rf/sub [:chats/raw-chat-messages-stream chat-id])
+        bottom-space 15]
     [rn/view
      {:style {:flex 1}}
-     ;;DO NOT use anonymous functions for handlers
+     ;; NOTE: DO NOT use anonymous functions for handlers
      [rn/flat-list
-      (merge
-       pan-responder
-       {:key-fn                       list-key-fn
-        :ref                          list-ref
-        :header                       [list-header chat]
-        :footer                       [list-footer chat]
-        :data                         messages
-        :render-data                  (get-render-data {:group-chat      group-chat
-                                                        :chat-id         chat-id
-                                                        :public?         public?
-                                                        :community-id    community-id
-                                                        :admins          admins
-                                                        :show-input?     show-input?
-                                                        :edit-enabled    true
-                                                        :in-pinned-view? false})
-        :render-fn                    render-fn
-        :on-viewable-items-changed    on-viewable-items-changed
-        :on-end-reached               list-on-end-reached
-        :on-scroll-to-index-failed    identity            ;;don't remove this
-        :content-container-style      {:padding-top    (+ bottom-space 32)
-                                       :padding-bottom 16}
-        :scroll-indicator-insets      {:top bottom-space} ;;ios only
-        :keyboard-dismiss-mode        :interactive
-        :keyboard-should-persist-taps :handled
-        :onMomentumScrollBegin        state/start-scrolling
-        :onMomentumScrollEnd          state/stop-scrolling
-        :scrollEventThrottle          16
-        :on-scroll                    on-scroll
-        ;;TODO https://github.com/facebook/react-native/issues/30034
-        :inverted                     (when platform/ios? true)
-        :style                        (when platform/android? {:scaleY -1})
-        :on-layout                    on-messages-view-layout})]
+      {:key-fn                       list-key-fn
```

Comment:
Not from your PR @yqrashawn, but I felt like sharing that the definition of `list-key-fn` with `defonce` is quite strange. It's actually the first time I've seen a definition like that in Clojure. It should be just a normal `defn`.

Reply:
fixed